client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -1165,6 +1207,24 @@ class CancelPurchaseForm extends Component {
 		return firstButtons.concat( isFirstStep ? [ next ] : [ prev, next ] );
 	};
 
+	fetchPurchaseExtendedStatus = async ( purchaseId ) => {
+		const res = await wpcom.req.get( {
```

Comment:
Not a blocker, but what happens if this endpoint fails? Should we add an error message?

Reply:
@sirbrillig Nothing happens even if the request fails since it only checks if the purchase is already extended with one month free offer. :)

Reply:
I meant, what happens to the render cycle if the network request itself fails? Say, if the user has a poor internet connection or there's some bug in the endpoint? It's hard to guarantee that network requests will never fail.

It seems to me that if it did fail it would cause `fetchPurchaseExtendedStatus` to throw an Error (since Promises throw when using async/await), which would then throw the error from `componentDidMount`. I'm not sure what happens then, but if it results in a blank white screen that might be poor UX for the user. We could wrap this in a `try/catch` either here or in `componentDidMount` to make it fail silently instead.

Reply:
@sirbrillig When I tested it, the component didn't result in a blank white screen even if the request failed and an error was thrown. That's why I said that nothing happens. However, it is always a good idea to make the code safer. I wrapped the request with a try-catch block as per your suggestion. Thanks for your review. :)